File: algorithm/3345. Smallest Divisible Digit Product I/solution.ts
function smallestNumber(n: number, t: number): number {
    
    for (let i = n; i <= 100; i++){        
        const listN = [...String(i)].map(num => Number(num));
        const product = listN.reduce((product, num) => product *= num, 1);

        if (product % t === 0) {
            return i;
        }
    }
};

File: algorithm/3345. Smallest Divisible Digit Product I/solution.java
class Solution {
    public int smallestNumber(int n, int t) {
        int out = -1;
        for (int i = n; i <= 100; i++) {
            String stringN = String.valueOf(i);
            char[] charArray = stringN.toCharArray();
            int product = 1;
            for (char c: charArray) {
                
                int num = Integer.parseInt(String.valueOf(c));
                product *= num;
            }

            if (product % t == 0) {
                out =  i;
                break;
            }
        }
        return out;
    }
}